homeassistant/components/solarlog/const.py

```diff
@@ -28,15 +28,15 @@
 class SolarLogSensorEntityDescription(SensorEntityDescription):
     """Describes Solarlog sensor entity."""
 
-    value: Callable[[float], int] = round
+    value: Callable[[float], float] = round
```

Comment:
You must adjust this to have the correct type hints, for example:
```suggestion
    value: Callable[[float | int | datetime], float | datetime] = round
```

Reply:
Thanks, but the issue I than have is that `round` can't be applied to `datetime`. Getting a lot of mypy errors than. 

![image](https://user-images.githubusercontent.com/24375847/146513701-f014c073-0723-4426-9868-2b1ce7471399.png)

What I actually try to achieve is, 
- by default, do nothing (something like `value=value`), 
- for some sensors I need some math to e.g. divide by 1000 and round to e.g. 3 digits. So I want to overwrite the default `value=value`
- and there is one `time` sensor, which needs to be told that the time stamp is in the local time zone.

It works as I have it now, but I agree the coding isn't as it should be. But I lack the experience to find out what to do to improve it. I hope you have some suggestions :-)

Reply:
Maybe something along those lines, which handles either a float conversion, or a datetime conversion, or None:
`value: Callable[[float], float] | Callable[[datetime], datetime] | None = None`

Reply:
Thanks, in combination with your last comment, this seem to work and makes sense.